src/http-requests.ts

```diff
@@ -20,16 +21,43 @@ function constructHostURL(host: string): string {
   }
 }
 
+function createHeaders(config: Config): Record<string, any> {
+  const agentHeader = 'X-Meilisearch-Client'
+  const contentType = 'Content-Type'
+  config.headers = config.headers || {}
+
+  const defaultHeaders = {
+    [agentHeader]: `Meilisearch JavaScript (v${PACKAGE_VERSION})`,
+    [contentType]: 'application/json',
+  }
+  const headers: Record<string, any> = {}
+
+  if (config.apiKey) {
+    headers['Authorization'] = `Bearer ${config.apiKey}`
+  }
+
+  // Creates the custom user agent with information on the package used.
+  if (config.clientAgents && Array.isArray(config.clientAgents)) {
+    const clients = config.clientAgents.concat(defaultHeaders[agentHeader])
+
+    headers[agentHeader] = clients.join(' ; ')
+  } else if (config.clientAgents && !Array.isArray(config.clientAgents)) {
+    // If the header is defined but not an array
+    throw new MeiliSearchError(
+      `Meilisearch: The header "${agentHeader}" should be an array of string(s).\n`
+    )
+  }
+
+  return { ...defaultHeaders, ...headers }
```

Comment:
I think you can join just the `contentType` header + everything else in the `headers` object.

Reply:
And make this `Meilisearch JavaScript (v${PACKAGE_VERSION})` a variable + remove the line 30.